vaadin-tabs-flow-parent/vaadin-tabs-flow/src/main/java/com/vaadin/flow/component/tabs/TabSheet.java

```diff
@@ -139,10 +141,7 @@ public Tab add(Tab tab, Component content, int position) {
             tabToContent.get(tab).removeFromParent();
         }
 
-        // On the client, content is associated with a tab by id
-        var id = "tabsheet-tab-" + UUID.randomUUID().toString();
-        tab.setId(id);
-        content.getElement().setAttribute("tab", id);
+        linkTabToContent(tab, content);
```

Comment:
What happens if the user changes a tab's id later?

Reply:
This case is also mentioned in the PR description